hathor/indexes/manager.py

```diff
@@ -35,13 +38,21 @@
 logger = get_logger()
 
 
+class _IndexFilter(Enum):
+    ALL = auto()  # block or tx, voided or not
+    VALID_BLOCKS = auto()  # only blocks that are not voided
+    VALID_TXS = auto()  # only transactions that are not voided
```

Comment:
What about soft-voided transactions?

Reply:
I'm not really sure about that actually. I'll have to review what timestamp-index and tips-index need for this to work correctly.

Reply:
Have you reviewed those?

Reply:
Not deeply enough, I'll add a test for testing how the indexes rebuild when voided txs exist, I think it might help me understand what actually happens.

Reply:
Done.